be/src/exec/hdfs_scanner.cpp

```diff
@@ -86,11 +88,33 @@ Status HdfsScanner::init(RuntimeState* runtime_state, const HdfsScannerParams& s
     _runtime_state = runtime_state;
     _scanner_params = scanner_params;
 
+    RETURN_IF_ERROR(_build_scanner_context());
```

Comment:
why move here?

Reply:
We need to build `_scanner_contex` and set `use_file_metacache` option before calling `do_init` where we may need to init `_cache`.